src/Polly.Core/ResilienceStrategy.cs

```diff
@@ -16,25 +16,29 @@ public abstract partial class ResilienceStrategy
     internal ResilienceStrategyOptions? Options { get; set; }
 
     /// <summary>
-    /// Executes the specified callback.
+    /// An implementation of resilience strategy that executes the specified <paramref name="callback"/>.
     /// </summary>
     /// <typeparam name="TResult">The type of result returned by the callback.</typeparam>
     /// <typeparam name="TState">The type of state associated with the callback.</typeparam>
     /// <param name="callback">The user-provided callback.</param>
     /// <param name="context">The context associated with the callback.</param>
     /// <param name="state">The state associated with the callback.</param>
-    /// <returns>An instance of <see cref="ValueTask"/> that represents an asynchronous callback.</returns>
+    /// <returns>
+    /// An instance of pending <see cref="ValueTask"/> for asynchronous executions or completed <see cref="ValueTask"/> task for synchronous exexutions.
+    /// </returns>
     /// <remarks>
-    /// This method is called by various methods exposed on <see cref="ResilienceStrategy"/>. These methods make sure that
-    /// <paramref name="context"/> is properly initialized with details about the execution mode.
+    /// <c>This method is called for both synchronous and asynchronous execution flows.</c>
```

Comment:
Did you mean to use `para` here rather than `c`?

Reply:
I intentionally used <c> to make the statement more "highlighted"

Reply:
Maybe `em` or `strong` would be better?

Reply:
Didn't know about <strong>, it's indeed better. Thanks :)